### ISSUE TITLE
`first` definition is extremely complicated

### ISSUE THREAD

**Issue by fadado** [jqlang/jq#1150]
In the file [builtin.jq](https://github.com/stedolan/jq/blob/master/src/builtin.jq) I find the definition of `limit` extremely complicated:

```
def first(g): label $out | foreach g as $item ([false, null]; if .[0]==true then break $out else [true, $item] end; .[1]) ;
```

I thik this one will do the work, or I'm missing something?

```
def first(g):  label $pipe | g | ., break $pipe ;
```

**Comment by pkoppstein** [comment]
Your definition is in fact semantically superior to the one in builtin.jq (see below) and I would certainly encourage you to submit a pull request, though I am not in a position to say it will be accepted.

(The current definition might have been written before the simpler definition was valid.)

To see why the simpler definition is significantly superior, compare:

````
jq -n 'first((1,2,3)|debug)'
["DEBUG:",1]
1
["DEBUG:",2]
````
````
jq -n 'def newfirst(g): label $pipe | g | ., break $pipe; newfirst((1,2,3)|debug)'
["DEBUG:",1]
1
````

**Comment by itchyny** [comment]
Already fixed in 1.6, resolved by 396543191cec32f69712b4364d26a0ebaa61b801 on Jan 31, 2017.

**Comment by itchyny** [comment]
The suggested solution was adopted in the master branch (as I commented before), thank you.
